test_conformance/compiler/test_build_helpers.cpp

```diff
@@ -1009,3 +1017,127 @@ REGISTER_TEST(get_program_build_info)
+                      "%d :\n%s! (%s from %s:%d)\n",
+                      length, ptr, IGetErrorString(error), __FILE__, __LINE__);
+            return TEST_FAIL;
+        }
+
+        clKernelWrapper kernel = clCreateKernel(program, name.c_str(), &error);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to create kernel with length of "
+                      "kernel name "
+                      "%d : %s! (%s from %s:%d)\n",
+                      length, name.c_str(), IGetErrorString(error), __FILE__,
+                      __LINE__);
+            return TEST_FAIL;
+        }
+
+        // query kernel name
+        size_t kernel_name_size = 0;
+        error = clGetKernelInfo(kernel, CL_KERNEL_FUNCTION_NAME, 0, nullptr,
+                                &kernel_name_size);
+        test_error(error, "clGetKernelInfo (size) failed");
+
+        std::vector<char> kernel_name(kernel_name_size);
+        error = clGetKernelInfo(kernel, CL_KERNEL_FUNCTION_NAME,
+                                kernel_name_size, kernel_name.data(), nullptr);
+        test_error(error, "clGetKernelInfo (name) failed");
+
+        if (name != std::string(kernel_name.data()))
+        {
+            log_error("Kernel name mismatch! expected=%s got=%s\n",
+                      name.c_str(), kernel_name.data());
+            return TEST_FAIL;
+        }
+    }
+
+#if defined(CL_VERSION_1_2)
+    {
+        clProgramWrapper programObj =
+            clCreateProgramWithSource(context, 1, &ptr, nullptr, &error);
+        test_error(error, "clCreateProgramWithSource failed (compile)");
+
+        error = clCompileProgram(programObj, 0, nullptr, nullptr, 0, nullptr,
+                                 nullptr, nullptr, nullptr);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to compile program with length of "
+                      "kernel name "
+                      "%d : %s! (%s from %s:%d)\n",
+                      length, name.c_str(), IGetErrorString(error), __FILE__,
+                      __LINE__);
+            return TEST_FAIL;
+        }
+
+        clProgramWrapper linkedProgram =
+            clLinkProgram(context, 0, nullptr, nullptr, 1, &programObj, nullptr,
+                          nullptr, &error);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to link program with length of "
+                      "kernel name "
+                      "%d : %s! (%s from %s:%d)\n",
+                      length, name.c_str(), IGetErrorString(error), __FILE__,
+                      __LINE__);
+            return TEST_FAIL;
+        }
+
+        clKernelWrapper kernel =
+            clCreateKernel(linkedProgram, name.c_str(), &error);
+        test_error(error, "clCreateKernel after link failed");
+    }
+#endif
+
+    return TEST_PASS;
+}
+
+REGISTER_TEST(kernel_name_size)
+{
+    cl_int status = TEST_PASS;
+    for (cl_uint len = 32; len < 2048; len *= 2)
```

Comment:
Check: do you want to run the 2K case, or stop at 1K?  Either way, it might be clearer to use `<=` here.

```suggestion
    for (cl_uint len = 32; len <= 2048; len *= 2)
```